src/main/java/io/shelang/aghab/resource/RedirectResource.java

```diff
@@ -0,0 +1,28 @@
+package io.shelang.aghab.resource;
+
+import io.quarkus.vertx.web.Route;
+import io.shelang.aghab.service.dto.LinksDTO;
+import io.shelang.aghab.service.link.LinksService;
+import io.smallrye.common.annotation.Blocking;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+import javax.inject.Inject;
+
+public class RedirectResource {
+
+    @Inject
+    LinksService linksService;
+
+    @Blocking
+    @Route(path = "/r/:hash", methods = HttpMethod.GET)
+    public void redirect(RoutingContext rc) {
+        LinksDTO byHash = linksService.getByHash(
+                rc.request().getParam("hash"));
+        rc.response()
+                .putHeader("location", byHash.getUrl())
+                .setStatusCode(301)
```

Comment:
By using 301 we redirect links **Permanently** and if a person clicks more than 1 time on the link the browser or other internet proxies may cache it. Do we what this happen? this has an effect on analytics, link redirection expiring, and disabling link.

Reply:
I think we could save redirection code in the database for each link too, what do you think for future updates

Reply:
> By using 301 we redirect links **Permanently** and if a person clicks more than 1 time on the link the browser or other internet proxies may cache it. Do we what this happen? this has an effect on analytics, link redirection expiring, and disabling links.  
> 

I think it's very good for high throughput, in my humble opinion we can set it optional and use 301 for normal links and 302 for the links that need an expiration, etc.

Reply:
> I think we could save redirection code in the database for each link too, what do you think for future updates

Could you please explain it more?

Reply:
> I think it's very good for high throughput, in my humble opinion we can set it optional and use 301 for normal links and 302 for the links that need an expiration, etc.

I agree with you, using both of them could help us.